src/rust/engine/fs/src/lib.rs

```diff
@@ -714,12 +837,13 @@ pub trait VFS<E: Send + Sync + 'static>: Clone + Send + Sync + 'static {
       return future::ok(vec![]).to_boxed();
     }
 
+    // let start = Instant::now();
     let init = PathGlobsExpansion {
       context: self.clone(),
-      todo: path_globs.include,
-      exclude: path_globs.exclude,
-      completed: HashSet::default(),
-      outputs: IndexMap::default(),
+      todo: IndexSet::from(path_globs.include.clone().into_iter().collect()),
```

Comment:
I'm not sure there is any advantage to treating this as a Set rather than a queue... we're already tracking completed entries using a set/map in `completed`, so this results in two tracking sets unnecessarily I think. Of course, I haven't measure the memory usage either way, but I know that the CPU usage will be higher.

Reply:
The utility of the IndexSet over a Vec would need to be profiled -- this is also an artifact from a previous implementation. I will change it to a Vec.

Reply:
(although, the reason why this was an optimization was because we potentially save on doing `directory_listing()`s which ping the fs if we have multiple of the same PathGlob to expand. but i think this is moot with the new changes)

Reply:
`directory_listings` are memoized in the Graph.

Reply:
That's great! I didn't realize that.